packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1499,7 +1499,9 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
     final _ModalScopeState<T>? scope = _scopeKey.currentState;
     assert(scope != null);
     for (final WillPopCallback callback in List<WillPopCallback>.of(_willPopCallbacks)) {
-      if (!await callback()) {
+      // TODO(goderbauer): Tests using the Component Framework in google3 insist on returning
```

Comment:
I'm guessing this is running in non-sound null safe mode?

Reply:
Probably. I am trying to chase down a contact person for the component framework to work out how to fix this. In the meantime, I am proposing this partial revert to unblock flutter rolls into g3.